### DISCUSSION TITLE
could i set a relative path for --icon or --add-data with --specpath [other directory] ?

### DISCUSSION THREAD

**Discussion by mrnextdoor0000** [pyinstaller/pyinstaller#6413] [PyInstaller] [answered]
I ran pyinstaller from python code like following.

```
import sys, os
import PyInstaller.__main__

args = [ ... ,
    '--icon', './myicon.ico',
    '--add-data', './resource;resource',
    '--workpath', './build',
    '--specpath', './build',
    '--distpath', './bin',
]

os.chdir(os.path.dirname(__file__))
PyInstaller.__main__.run(args)
```

Then i failed and got a message `Unable to find "build\resource" when adding binary and data files.`
A freezing succeeded when i removed the option `['--specpath', './build']`
I guess that pyinstaller use the specpath as a root directory to search data files.
Similar things occurred with --icon option, building failed with --specpath.
Can I use relative path relative to script file not specfile..?
or Can I specify a root directory for icon, data files or binary files..?
I am using python 397 and pyinstaller 4.7.

**Comment by bwoodsend** [comment]
And where is your Python code and icon?

**Comment by rokm** [comment, marked as answer]
No, relative resource paths are always treated as relative to the location of the spec file.

But in your example, you could easily transform script-relative paths to spec-file-relative paths or even absolute paths before passing them to PyInstaller.

**Reply by mrnextdoor0000** [reply]
Thanks for the clear answer. I will follow your advice.
